### PR TITLE
Add standard batch functions

### PR DESCRIPTION
**Pull request by giarc3**
Adds `encrypt_batch`, `decrypt_batch`, and `encrypt_with_existing_edek_batch`

### CODE REVIEW

File: src/saas_shield/standard.rs [hunk 1]
```diff
@@ -277,6 +404,59 @@ impl StandardDocumentOps for SaasShieldStandardClient {
...
+            )
+        };
+        let decryption_result = collection_to_batch_result(docs_and_keys, encrypt_document);
+        let combined_failures = batch_unwrap_response
+            .failures
+            .into_iter()
+            .map(|(k, err)| (k, err.into()))
+            .chain(edek_failures)
+            .chain(decryption_result.failures)
+            .collect();
```
Comment (skeet70) [inline src/saas_shield/standard.rs:453]: It seems like we've done this same pattern for merging failures several times now. Could we make a trait that allows it or a util function `merge_batch_failures` or something?

File: src/standard.rs [hunk 2]
```diff
@@ -130,6 +152,12 @@ pub trait StandardDocumentOps: AlloyClient {
         plaintext_document: PlaintextDocumentWithEdek,
         metadata: &AlloyMetadata,
     ) -> Result<EncryptedDocument, AlloyError>;
+    /// TODO
```
Comment (skeet70) [inline src/standard.rs:155]: TODO in another linked ticket or still in this one?

**Review by coltfred — COMMENTED** [review]
Just a couple comments. I know you're doing a big refactor on the newtypes, etc.

Just wanted to get you these so they could be part of the fun. 😄

File: src/lib.rs [hunk 3]
```diff
@@ -35,6 +35,7 @@ pub mod vector;
 uniffi::setup_scaffolding!();
 
 type FieldId = String;
+type DocumentId = String;
```
[the remark this answers is not in the document]
Comment (giarc3) [reply src/lib.rs]: I changed all the public aliases newtype structs

File: src/saas_shield/standard.rs [hunk 4]
```diff
@@ -169,6 +170,36 @@ impl SaasShieldStandardClient {
...
+    {
+        let request_metadata = metadata.clone().try_into()?;
+        let decompose_edek = |edek_with_header: EdekWithKeyIdHeader| {
+            let edek_parts = Self::decompose_edek_header(edek_with_header)?;
+            edek_parts.get_edek_bytes()
+        };
+        let BatchResult {
+            successes: edeks,
+            failures: edek_failures_str,
+        } = collection_to_batch_result(edeks_with_headers, decompose_edek);
```
[the remark this answers is not in the document]
Comment (giarc3) [reply src/saas_shield/standard.rs]: Sure, done

File: src/saas_shield/standard.rs (hunk 1 above)
Comment (giarc3) [reply src/saas_shield/standard.rs]: I looked into this and couldn't find a nice way to do it. The pattern is used in 3 places and is slightly different among each, so I think I'd rather leave it for now

File: kotlin/src/test/kotlin/com/ironcorelabs/ironcore_alloy/IroncoreAlloyTest.kt [hunk 5]
```diff
@@ -179,6 +179,21 @@ class IroncoreAlloyTest {
...
 
+    @Test
+    fun sdkStandardBatchRoundtrip() {
+        val plaintextDocument = mapOf("foo" to "My data".toByteArray())
+        val plaintextDocuments = mapOf("doc" to plaintextDocument)
+        val metadata = AlloyMetadata.newSimple("tenant")
+        runBlocking {
+            val encrypted = sdk.standard().encryptBatch(plaintextDocuments, metadata)
+            assertEquals(encrypted.successes.size, 1)
+            assertEquals(encrypted.failures.size, 0)
```
[the remark this answers is not in the document]
Comment (giarc3) [reply kotlin/src/test/kotlin/com/ironcorelabs/ironcore_alloy/IroncoreAlloyTest.kt:190]: It felt like it was covered by other tests/languages, but I can add it if you'd prefer

File: src/standard.rs (hunk 2 above)
Comment (giarc3) [reply src/standard.rs]: This one. Done

**Review by skeet70 — APPROVED** [review]

**Review by coltfred — APPROVED** [review]
PR #100!

**Pull request merged**